### PR TITLE
Refactor some MFA pages, continue wiring to backend

### PR DESCRIPTION
**Pull request by nickclyde**
## Related Issue or Background Info

- This continues the account creation work, wiring the frontend MFA pages into a flow, and hitting backend endpoints when needed
- Depends on #1664 

## Changes Proposed

- Does some refactoring to reduce duplication across similar MFA pages
- Add lots more service methods, and refactor the service to reduce duplication

## Additional Information

- These changes should be sufficient along with #1664 to successfully go through the entire flow for SMS, Email, and Voice call MFA

### UI
- [x] Any changes to the UI/UX are approved by design 
- [x] Any new or updated content (e.g. error messages) are approved by design 

### Testing
- [ ] Includes a summary of what a code reviewer should verify

### Changes are Backwards Compatible
- [ ] Database changes are submitted as a separate PR
  - [ ] Any new tables that do not contain PII are accompanied by a GRANT SELECT to the no-PHI user
  - [ ] Any changes to tables that have custom no-PHI views are accompanied by changes to those views
- [ ] GraphQL schema changes are backward compatible with older version of the front-end

### Security
- [ ] Changes with security implications have been approved by a security engineer (changes to  authentication, encryption, handling of PII, etc.)
- [ ] Any dependencies introduced have been vetted and discussed

## Cloud
- [ ] DevOps team has been notified if PR requires ops support
- [ ] If there are changes that cannot be tested locally, this has been deployed to our Azure `test` environment for verification

### CODE REVIEW

File: frontend/src/app/accountCreation/MfaSmsVerify/MfaSmsVerify.tsx
Comment (emmastephenson) [inline frontend/src/app/accountCreation/MfaSmsVerify/MfaSmsVerify.tsx:13]: Do we need to give the user a button to re-request the activation code for all of these verification pages? The endpoint for that is `/resend-activation-passcode`. (Note that it's not at all written on the backend yet, lol, so if you want to wait to add it to these verification pages that's totally fine. It might just be a good idea to include a TODO)

**Review by emmastephenson — COMMENTED** [review]
Looks fantastic, thanks for all your work on this! My only comment is about re-requesting the activation code.

File: frontend/src/app/accountCreation/MfaEmailVerify/MfaEmailVerify.tsx
Comment (nickscialli-usds) [inline frontend/src/app/accountCreation/MfaEmailVerify/MfaEmailVerify.tsx:12]: Is this expiry time provided by the backend (or okta) at all? Might be worth wiring it in here in case it changes. Or it might not be worth the effort 🤷‍♂️

File: frontend/src/app/accountCreation/MfaPhoneVerify/MfaPhoneVerify.tsx
Comment (nickscialli-usds) [inline frontend/src/app/accountCreation/MfaPhoneVerify/MfaPhoneVerify.tsx:12]: Same issue: seems like this prop is guaranteed. Also, it would look pretty gnarly if it defaulted to the empty string

File: frontend/src/app/accountCreation/MfaSelect/MfaSelect.tsx
Comment (nickscialli-usds) [inline frontend/src/app/accountCreation/MfaSelect/MfaSelect.tsx:46]: Can you make the `mfaOption` better-typed (as a string union) if we know these are the options?

File: frontend/src/app/accountCreation/MfaSendCodeToContact/MfaSendCodeToContact.tsx
Comment (nickscialli-usds) [inline frontend/src/app/accountCreation/MfaSendCodeToContact/MfaSendCodeToContact.tsx:61]: This error is the entire response object, right? So will this contact error look like "API Error: [object Object]"?

File: frontend/src/app/accountCreation/MfaSendCodeToContact/MfaSendCodeToContact.tsx
Comment (nickscialli-usds) [inline frontend/src/app/accountCreation/MfaSendCodeToContact/MfaSendCodeToContact.tsx:72]: This effect probably runs too much given there's no dependency array. I think we should have `formIsDirty` and `validateContact` in the dependency array. You might have to wrap the `validateContact` function in a `useCallback` hook to not run the hook excessively (and make linting errors go away)

File: frontend/src/app/accountCreation/MfaVerify/MfaVerify.tsx
Comment (nickscialli-usds) [inline frontend/src/app/accountCreation/MfaVerify/MfaVerify.tsx:93]: Is this handled in a future piece of work?

File: frontend/src/app/accountCreation/AccountCreationApiService.ts
Comment (nickclyde) [reply frontend/src/app/accountCreation/AccountCreationApiService.ts:17]: Yeah, I think it should work without that, I'll remove

File: frontend/src/app/accountCreation/MfaEmailVerify/MfaEmailVerify.tsx
Comment (nickclyde) [reply frontend/src/app/accountCreation/MfaEmailVerify/MfaEmailVerify.tsx:11]: I think I added the optional chaining before I realized that it should actually be required. So I think it's okay for me to get rid of the optional chaining and the empty string :+1:

File: frontend/src/app/accountCreation/MfaEmailVerify/MfaEmailVerify.tsx
Comment (nickclyde) [reply frontend/src/app/accountCreation/MfaEmailVerify/MfaEmailVerify.tsx:12]: Hmmm, it's possibly provided by Okta, but I think it just defaults to 10 minutes and I don't know if we plan on changing it ever

File: frontend/src/app/accountCreation/MfaPhoneVerify/MfaPhoneVerify.tsx
Comment (nickclyde) [reply frontend/src/app/accountCreation/MfaPhoneVerify/MfaPhoneVerify.tsx:12]: Yep, should be guaranteed, will remove this as well

File: frontend/src/app/accountCreation/MfaSelect/MfaSelect.tsx
Comment (nickclyde) [reply frontend/src/app/accountCreation/MfaSelect/MfaSelect.tsx:46]: I'm on it!

File: frontend/src/app/accountCreation/MfaSendCodeToContact/MfaSendCodeToContact.tsx
Comment (nickclyde) [reply frontend/src/app/accountCreation/MfaSendCodeToContact/MfaSendCodeToContact.tsx:61]: Yeah, good catch. I don't actually know what the error object will look like though. I suppose `error.message` is a good bet?

File: frontend/src/app/accountCreation/MfaVerify/MfaVerify.tsx
Comment (nickclyde) [reply frontend/src/app/accountCreation/MfaVerify/MfaVerify.tsx:93]: This is related to Emma's comment above - I'm gonna go ahead and implement it as part of this PR.

File: frontend/src/app/accountCreation/MfaEmailVerify/MfaEmailVerify.tsx
Comment (emmastephenson) [reply frontend/src/app/accountCreation/MfaEmailVerify/MfaEmailVerify.tsx:12]: https://support.okta.com/help/s/question/0D50Z00008G7VcR/how-long-are-sms-one-time-passcode-valid?language=en_US

I think it actually defaults to 5 minutes, and it doesn't look like we can change that :/

Comment (nickclyde) [reply frontend/src/app/accountCreation/MfaEmailVerify/MfaEmailVerify.tsx:12]: Aha! I will update the copy to 5  minutes then

File: frontend/src/app/accountCreation/MfaSmsVerify/MfaSmsVerify.tsx
Comment (nickclyde) [reply frontend/src/app/accountCreation/MfaSmsVerify/MfaSmsVerify.tsx]: I ended up implementing the service function and attaching it to the "Send another code" button in `MfaVerify` :+1:

File: frontend/src/app/accountCreation/MfaEmailVerify/MfaEmailVerify.tsx
Comment (nickclyde) [reply frontend/src/app/accountCreation/MfaEmailVerify/MfaEmailVerify.tsx]: Fixed in 64b96ec

File: frontend/src/app/accountCreation/MfaSendCodeToContact/MfaSendCodeToContact.tsx
Comment (cdevent) [inline frontend/src/app/accountCreation/MfaSendCodeToContact/MfaSendCodeToContact.tsx:45]: Any reason why you do like that but not like this: 
```suggestion
    const valid = contactIsValid(contact);
```

Comment (nickclyde) [reply frontend/src/app/accountCreation/MfaSendCodeToContact/MfaSendCodeToContact.tsx:45]: Yes, because I want to catch any exceptions thrown by `contactIsValid`, and just set valid to `false` in that case. If I accept your change, any exceptions thrown will crash the app

Comment (cdevent) [reply frontend/src/app/accountCreation/MfaSendCodeToContact/MfaSendCodeToContact.tsx:45]: I've looked at all functions that used here and didn't see any exceptions in any case you receive boolean value.

Comment (nickclyde) [reply frontend/src/app/accountCreation/MfaSendCodeToContact/MfaSendCodeToContact.tsx:45]: That may be the case, but _just in case_ anything funky happens with phone or email validation, I would rather it be the case that the app doesn't crash, and instead valid gets set to false. I don't think it hurts anything to keep the try catch?

**Review by emmastephenson — APPROVED** [review]
LGTM, great work! Let me know if you want to pair to make sure this works in `test` one last time before you merge.

**Comment by emmastephenson** [comment]
Oh, and just a heads-up that the passcode verification won't work until #1751 is submitted. You can definitely merge this PR before that one though, because the endpoint is there, it just won't do anything until #1751 is in.

**Pull request merged**
